tests/services/logging.service.spec.ts

```diff
@@ -1,64 +1,160 @@
+        return key in mockLocalStorage ? mockLocalStorage[key] : null;
+      },
+      setItem: (key: string, value: string) => {
+        mockLocalStorage[key] = value;
+      },
+      removeItem: (key: string) => {
+        delete mockLocalStorage[key];
+      },
+      clear: () => {
+        mockLocalStorage = {};
+      }
+    };
+
+    // Mock function calls
+    spyOn(localStorage, 'getItem').and.callFake(mockLocalStorageFunctions.getItem);
+    spyOn(localStorage, 'setItem').and.callFake(mockLocalStorageFunctions.setItem);
+    spyOn(localStorage, 'removeItem').and.callFake(mockLocalStorageFunctions.removeItem);
+    spyOn(localStorage, 'clear').and.callFake(mockLocalStorageFunctions.clear);
+
+    // Mock dates
+    jasmine.clock().install();
+    jasmine.clock().mockDate(mockDate);
+
+    // Initialize logging service
     const electronService = jasmine.createSpyObj('ElectronService', ['isElectron']);
     electronService.isElectron = jasmine.createSpy('isElectron', () => false);
     loggingService = new LoggingService(electronService);
+    headerLog = `${loggingService.LOG_START_HEADER}\n${mockDate.toLocaleString()}`;
+    loggingService.reset();
+    localStorage.clear();
   });
 
-  describe('.getTrimmedLogCache()', () => {
-    it('should return 1 new log if cache does not contain existing log', () => {
-      expect(getFilteredLogCount(undefined, logHeaderFilter)).toEqual(1);
-      expect(getFilteredLogCount('', logHeaderFilter)).toEqual(1);
-      expect(getFilteredLogCount('gibberish', logHeaderFilter)).toEqual(1);
+  afterEach(() => {
+    loggingService.reset();
+    localStorage.clear();
+    jasmine.clock().uninstall();
+  });
+
+  describe('.startSession()', () => {
+    it('should successfully initialize logging session', () => {
+      loggingService.startSession();
+      const actualLog = loggingService.getCachedLog();
+      const expectedLog = headerLog;
+      expect(actualLog).toEqual(expectedLog);
+    });
+
+    it('should successfully reinitialize logging session', () => {
+      loggingService.startSession();
+      loggingService.reset();
+      loggingService.startSession();
+      const actualLog = loggingService.getCachedLog();
+      const expectedLog = `${headerLog}${sessionSeparator}${headerLog}`;
+      expect(actualLog).toEqual(expectedLog);
+    });
+
+    it('should successfully reinitialize logging session when limit reached', () => {
+      for (let i = 0; i < loggingService.LOG_COUNT_LIMIT; i += 1) {
+        loggingService.startSession();
+        loggingService.reset();
+      }
+      loggingService.startSession();
+      const actualLog = loggingService.getCachedLog();
+      let expectedLog = '';
+      for (let i = 0; i < loggingService.LOG_COUNT_LIMIT; i += 1) {
+        if (i === loggingService.LOG_COUNT_LIMIT - 1) {
+          expectedLog += headerLog;
+        } else {
+          expectedLog += `${headerLog}${sessionSeparator}`;
+        }
+      }
```

Comment:
```suggestion
      const expectedLog = [...Array(loggingService.LOG_COUNT_LIMIT)].map( _ => headerLog).join(sessionSeparator);
```
A declarative style would definitely be better here

Reply:
It definitely looks more concise and clear, thanks for the suggestion!